lib/ConsumerImpl.cc

```diff
@@ -319,6 +320,27 @@ void ConsumerImpl::unsubscribeAsync(ResultCallback originalCallback) {
     }
 }
 
+void ConsumerImpl::triggerCheckExpiredChunkedTimer() {
+    checkExpiredChunkedTimer_ = executor_->createDeadlineTimer();
+    checkExpiredChunkedTimer_->expires_from_now(
+        boost::posix_time::milliseconds(expireTimeOfIncompleteChunkedMessageMs_));
+    checkExpiredChunkedTimer_->async_wait([this](const boost::system::error_code& ec) -> void {
```

Comment:
Need to capture `smart pointers` to prevent segmentation faults. Weak pointers are recommended to prevent circular references.
https://github.com/apache/pulsar/pull/17481